src/service.rs

```diff
@@ -0,0 +1,388 @@
+#![allow(clippy::cognitive_complexity)]
```

Comment:
So right now this file seems to have most of the old `main.rs` content. I'd prefer that it go back there and we have this file include most of the `vector_windows.rs` code. It'd be particularly awesome if we can have it structured so that we could add other OS's later in a very clean way...

Reply:
I extracted parts of this file because I needed to parse command-line in `vector_windows.rs` when starting as a service.
I wanted to keep windows specific code and boilerplate stuff separated in its own source file.

 So what you say is that content of this file should be directory put in `main.rs`, inside a `#[cfg(windows)]` directive ?

We could work on some interface that would make it easy to add support for other OS's later. That's what I started to do initially before giving up because Linux is already supported through `systemd`. 

`telegraf` uses this [library](https://github.com/kardianos/service) that supports Windows XP+, Linux, OSX

Reply:
Hmmm yeah I think we had the same feeling here. Ok!

Reply:
What do you think we should do about `main.rs` ? Moving code back or keeping things as they are right now ?